crates/stc_ts_file_analyzer/tests/pass/types/mapped/recursiveMappedTypes/1.ts

```diff
@@ -0,0 +1,19 @@
+// Repro from #29992
+
+type NonOptionalKeys<T> = {
+  [P in keyof T]: undefined extends T[P] ? never : P;
+}[keyof T];
+type Child<T> = { [P in NonOptionalKeys<T>]: T[P] };
+
+export interface ListWidget {
+  type: "list";
+  minimum_count: number;
+  maximum_count: number;
+  collapsable?: boolean; //default to false, means all expanded
+  each: Child<ListWidget>;
```

Comment:
This test file should not stored in `pass`.

https://www.typescriptlang.org/play?#code/PTAECUFMAcCcHtQDMEFtQGIBMBOPWAoAgFwE9pJQA5eAOwHlpiBLOgQwBsBpSUgZwA8AFQB8oALygA3gVCgA2gAVQzWqADWveElBCAugC5QAV1oATSElWQzoSAA9ikc311K9oAPyhakAG6QsKBGigDcBAC+8pqk2rp64WQUoADCABbMHGbCYpJSCsqq1HSMLOzcvIKihm6KHhHhBA7Q8LDEKrROsEhsAMaUADLMfMQA6sxmAOaQ7TJySZBGAEQcw8RL4XKoqsyoxqgA+r3wpsRGtPsARoGboKhs9rv7Ryed51c3sqDHHBxs0Hw2JcOJBPEZLvB4CC2LRQqAQBYesYOO1iIgehw+JAADR3SAw1ycDh2ezQGEWMxffG9NJGdKZbJDEbjKYzEThCJEBagJnEelZCSpDJZAS8lnTYjsogWXp-WCUEHtexGXn8szhewAOgW4SAA

Reply:
If you put this in the error box, you won't pass the test.
I don't have an error with the circulation reference.

Why don't you just ignore it?

Reply:
I will append `.` at file name

Reply:
You should move it to `visualize` instead

Reply:
what is mean `visualize test`?

Reply:
file analyzer/tests/visualize

Reply:
I was just wondering,
What does the test mean?

Reply:
https://github.com/dudykr/stc/tree/d3d6a8fa98f5d6cceefb046124fcf6b63d9305cb/crates/stc_ts_file_analyzer#tests

Reply:
My vision was dark. I'm so sorry 
I fixed its